presto-spi/src/main/java/com/facebook/presto/spi/block/ClosingBlockLease.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.block;
+
+import com.facebook.presto.spi.api.Experimental;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.unmodifiableList;
+import static java.util.Objects.requireNonNull;
+
+@Experimental
+@NotThreadSafe
+public final class ClosingBlockLease
+        implements BlockLease
+{
+    private final Block block;
+    private final List<Closer> closers;
+    private boolean closed;
+    private boolean retrieved;
+
+    public ClosingBlockLease(Block block, Closer... closers)
+    {
+        requireNonNull(block, "block is null");
+        requireNonNull(closers, "closers is null");
+        this.block = block;
+        this.closers = closers.length == 0 ? emptyList() : unmodifiableList(asList(closers.clone()));
+    }
+
+    public static BlockLease newLease(Block block, Closer... closers)
```

Comment:
nit: Current use case all only needs one `Closer`? Why it's a `...` ? 

Also, why not taking a `List<Closer>` as input ?